tests/countries/test_nepal.py

```diff
@@ -0,0 +1,408 @@
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Maha Ashtami"
+        dt = (
+            "2019-10-06",
+            "2020-10-23",
+            "2021-10-13",
+            "2022-10-03",
+            "2023-10-22",
+            "2024-10-10",
+            "2025-09-30",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Maha Navami"
+        dt = (
+            "2019-10-07",
+            "2020-10-24",
+            "2021-10-14",
+            "2022-10-04",
+            "2023-10-23",
+            "2024-10-11",
+            "2025-10-01",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Maghe Sankranti"
+        dt = (
+            "2019-01-15",
+            "2020-01-15",
+            "2021-01-14",
+            "2022-01-14",
+            "2023-01-14",
+            "2024-01-14",
+            "2025-01-14",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Mha Puja"
+        dt = (
+            "2019-10-28",
+            "2020-11-15",
+            "2021-11-05",
+            "2022-10-25",
+            "2023-11-13",
+            "2024-11-02",
+            "2025-10-22",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Nepali New Year (Vikram Sambat)"
+        dt = (
+            "2019-04-14",
+            "2020-04-13",
+            "2021-04-14",
+            "2022-04-14",
+            "2023-04-14",
+            "2024-04-13",
+            "2025-04-13",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Phulpati"
+        dt = (
+            "2019-10-05",
+            "2020-10-22",
+            "2021-10-12",
+            "2022-10-02",
+            "2023-10-21",
+            "2024-10-09",
+            "2025-09-29",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Ram Navami"
+        dt = (
+            "2019-04-13",
+            "2020-04-02",
+            "2021-04-21",
+            "2022-04-10",
+            "2023-03-30",
+            "2024-04-17",
+            "2025-04-06",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Sonam Losar"
+        dt = (
+            "2019-02-05",
+            "2020-01-25",
+            "2021-02-12",
+            "2022-02-02",
+            "2023-01-22",
+            "2024-02-10",
+            "2025-01-30",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Tihar Holiday"
+        dt = (
+            "2019-10-30",
+            "2020-11-17",
+            "2021-11-07",
+            "2022-10-27",
+            "2023-11-15",
+            "2024-11-04",
+            "2025-10-24",
+        )
+        self.assertHolidayName(name, dt)
+
+        name = "Vijayadashami"
+        dt = (
+            "2019-10-08",
+            "2020-10-25",
+            "2021-10-15",
+            "2022-10-05",
+            "2023-10-24",
+            "2024-10-12",
+            "2025-10-02",
+        )
+        self.assertHolidayName(name, dt)
+
```

Comment:
As for test for holidays categories - see [`test_macau.py`](https://github.com/vacanza/holidays/blob/ee26941bcfccd9028a85113f129ff1206b7e2071/tests/countries/test_macau.py#L821) and [`test_taiwan.py`](https://github.com/vacanza/holidays/blob/ee26941bcfccd9028a85113f129ff1206b7e2071/tests/countries/test_taiwan.py#L650) examples, both approach works so just use the ones you feel works best for you 👍